### ISSUE TITLE
"Can't convert Rufus::Scheduler::ZoTime into an exact number" with rufus-scheduler 3.3.0

### ISSUE THREAD

**Issue by turley** [ondrejbartas/sidekiq-cron#120]
When using sidekiq-cron with the latest version of rufus-scheduler (3.3.0), I'm getting the following error reported for each of my jobs:
```
ERROR: CRON JOB: can't convert Rufus::Scheduler::ZoTime into an exact number
ERROR: CRON JOB: /Users/ben/.rvm/gems/ruby-2.3.0/gems/activesupport-5.0.0.1/lib/active_support/core_ext/time/calculations.rb:249:in `-'
```

I haven't looked into the underlying cause yet (it could be a bug in rufus-scheduler), but I did find that the errors can be avoided by using an earlier version of rufus-scheduler (3.2.2).

**Comment by ondrejbartas** [comment]
@turley Hi, can you share your configuration of cron line?

**Comment by turley** [comment]
@ondrejbartas It happens with any/all of my various job configurations, but just to confirm, I removed all the others and reproduced the issue with the following:
```
Sidekiq::Cron::Job.create(name: 'my_cron_job', cron: '*/5 * * * *', class: 'MyWorker')
```

**Comment by turley** [comment]
Just FYI: I'm *not* using ActiveJob/Rails. I am using sidekiq 4.2.6 and activesupport 5.0.0.1 with the latest revision of sidekiq-cron (master@3b057de).

**Comment by turley** [comment]
I've found what I believe to be the problem:

In a few places (e.g. https://github.com/ondrejbartas/sidekiq-cron/blob/v0.4.4/lib/sidekiq/cron/job.rb#L546), sidekiq-cron is calling methods like `#previous_time` and `#next_time` on an instance of Rufus::Scheduler::CronLine.

In rufus-scheduler 3.2.2, those methods actually returned instances of `Time` 
(https://github.com/jmettraux/rufus-scheduler/blob/v3.2.2/lib/rufus/scheduler/cronline.rb#L186) which could be used in time calculations (`-` and `+` methods), but in rufus-scheduler 3.3.0, those methods are returning `ZoTime` objects (https://github.com/jmettraux/rufus-scheduler/blob/v3.3.0/lib/rufus/scheduler/cronline.rb#L200).

If rufus-scheduler is still intending those methods to return `Time` objects, I'd say that's probably a bug in rufus-scheduler. Otherwise, I think you'd have to handle it here and call `#to_time`  appropriately when a `ZoTime` instance is returned (perhaps using `respond_to?(:to_time)`).

**Comment by ondrejbartas** [comment]
thanks for debugging, for now you can specify version of rufus-scheduler in your gemfile and sidekiq-cron will use correct one, will try to fix this in near future

**Comment by Junyulive** [comment]
Fixed in #121

**Comment by ondrejbartas** [comment]
thanks @Junyulive

**Comment by turley** [comment]
Thanks - that fixes the issue for me.

**Comment by mironov** [comment]
@ondrejbartas Could you publish the new version to rubygems?

**Comment by ondrejbartas** [comment]
@mironov released 0.4.5
